File: src/test/java/se/johannesdahlgren/adventofcode2021/Day14Test.java
package se.johannesdahlgren.adventofcode2021;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class Day14Test {

  @Test
  void part1Example() {
    Day14 day14 = new Day14("day14Example.txt");
    long value = day14.mostCommonMinusLeastCommonElementAfter(10);
    assertThat(value).isEqualTo(1588L);
  }

  @Test
  void part1() {
    Day14 day14 = new Day14("day14.txt");
    long value = day14.mostCommonMinusLeastCommonElementAfter(10);
    assertThat(value).isEqualTo(2621L);
  }
}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day14Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

class Day14Test {

    @Test
    fun part1Example() {
        val day14 = Day14("day14Example")
        assertThat(day14.calcUnitOfSandAtRest(0)).isEqualTo(24)
    }

    @Test
    fun part1() {
        val day14 = Day14("day14")
        assertThat(day14.calcUnitOfSandAtRest(0)).isEqualTo(737)
    }

    @Test
    fun part2Example() {
        val day14 = Day14("day14Example")
        assertThat(day14.calcUnitOfSandAtRest(2)).isEqualTo(93)
    }

    @Disabled("Slow, around 2 minutes")
    @Test
    fun part2() {
        val day14 = Day14("day14")
        assertThat(day14.calcUnitOfSandAtRest(2)).isEqualTo(28145)
    }
}